File: java/11CompleteImplementationCodeRefactored/src/main/java/com/neomatrix369/domain/WordEntity.java
package com.neomatrix369.domain;

public class WordEntity {
    private String word;
    private int frequency;

    public WordEntity(String word, int frequency) {
        this.word = word;
        this.frequency = frequency;
    }

    public String getWord() {
        return word;
    }

    public int getFrequency() {
        return frequency;
    }

    public void incrementFrequency() {
        frequency++;
    }

    public boolean isTheSameAs(String word) {
        return this.word.equals(word);
    }
}

File: csharp/11CompleteImplementationCode/WordEntity.cs
using System;

namespace CompleteImplementationCode
{
	public class WordEntity
	{
		public string Word { get; set; }

		public int Frequency { get; set; }

		public void IncrementFrequency() {
			Frequency++;
		}

		public bool isTheSameAs(String word) {
			return Word == word;
		}
	}
}
